model_utilities/train/data_loader.py

```diff
@@ -19,3 +19,33 @@ def load_dataset(train_path: str, validation_path: str, test_path: str) -> ds.Da
     dataset = dataset.remove_columns("Unnamed: 0")
 
     return dataset
+
+
+def prepare_train_dataset(config: dict) -> ds.DatasetDict:
+    weights = {item[0]: float(item[1]) for item in config.items("DATASET WEIGHTS")}
+
+    paths = {key: {word: os.path.join(config["DIRECTORIES"]["preprocessed_data_dir"], f"{word}-{key}.tsv") for word in ["train", "val"]} for key in weights.keys()}
+
+    datasets = {key: load_dataset(paths[key]["train"], paths[key]["val"]) for key in paths}
+
+    # Need to interleave each split separately
+    train_datasets = {key: datasets[key]["train"] for key in datasets}
+
+    train_dataset_combined = ds.interleave_datasets(list(train_datasets.values()),
+                                                    list(weights.values()),
+                                                    seed=int(config["TRAINING"]["shuffle_seed"]),
+                                                    stopping_strategy="all_exhausted",
+                                                    )
+    val_datasets = {key: datasets[key]["validation"] for key in datasets}
+
+    val_dataset_combined = ds.interleave_datasets(list(val_datasets.values()),
```

Comment:
```suggestion
    val_dataset_combined = ds.interleave_datasets(list(val_datasets.values()),
```
The term `val` can be misleading as it implies `value`, which it is not.